x-pack/plugins/monitoring/server/lib/setup/collection/get_collection_status.js

```diff
@@ -0,0 +1,313 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get, uniq } from 'lodash';
+import { METRICBEAT_INDEX_NAME_UNIQUE_TOKEN } from '../../../../common/constants';
+import { KIBANA_SYSTEM_ID, BEATS_SYSTEM_ID, LOGSTASH_SYSTEM_ID, APM_SYSTEM_ID } from '../../../../../xpack_main/common/constants';
+
+const getRecentMonitoringDocuments = async (req, indexPatterns, clusterUuid) => {
+  const start = get(req.payload, 'timeRange.min', 'now-30s');
+  const end = get(req.payload, 'timeRange.max', 'now');
+
+  const filters = [
+    {
+      range: {
+        'timestamp': {
+          gte: start,
+          lte: end
+        }
+      }
+    }
+  ];
+
+  if (clusterUuid) {
+    filters.push({ term: { 'cluster_uuid': clusterUuid } });
+  }
+
+  const params = {
+    index: Object.values(indexPatterns),
+    size: 0,
+    ignoreUnavailable: true,
+    filterPath: [
+      'aggregations.indices.buckets'
+    ],
+    body: {
+      query: {
+        bool: {
+          filter: filters,
+        }
+      },
+      aggs: {
+        indices: {
+          terms: {
+            field: '_index',
+            size: 50,
+          },
+          aggs: {
+            es_uuids: {
+              terms: {
+                field: 'node_stats.node_id'
+              },
+              aggs: {
+                by_timestamp: {
+                  max: {
+                    field: 'timestamp'
+                  }
+                }
+              }
+            },
+            kibana_uuids: {
+              terms: {
+                field: 'kibana_stats.kibana.uuid'
+              },
+              aggs: {
+                by_timestamp: {
+                  max: {
+                    field: 'timestamp'
+                  }
+                }
+              }
+            },
+            beats_uuids: {
+              terms: {
+                field: 'beats_stats.beat.uuid'
+              },
+              aggs: {
+                by_timestamp: {
+                  max: {
+                    field: 'timestamp'
+                  }
+                }
+              }
+            },
+            logstash_uuids: {
+              terms: {
+                field: 'logstash_stats.logstash.uuid'
+              },
+              aggs: {
+                by_timestamp: {
+                  max: {
+                    field: 'timestamp'
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  };
+
+  const { callWithRequest } = req.server.plugins.elasticsearch.getCluster('monitoring');
+  return await callWithRequest(req, 'search', params);
+};
+
+async function detectProducts(req) {
+  const result = {
+    [KIBANA_SYSTEM_ID]: {
+      doesExist: true,
+    },
+    'elasticsearch': {
+      doesExist: true,
+    },
+    [BEATS_SYSTEM_ID]: {
+      doesExist: false,
+      mightExist: false,
+    },
```

Comment:
What's the difference between `doesExist` and `mightExist` for Beats, APM, Logstash? Under what scenario would we set `doesExist` to `true`?

Reply:
Good question. Right now, I don't know if there is a way to _definitively_ know that some system exists and should be monitored. I left both options as a way to future proof moving forward, but it might _only_ be `mightExist` for these. Thoughts?

Reply:
Yeah, I think just `mightExist` for these products make sense conceptually, at least for now.